tool/tsh/proxy.go

```diff
@@ -448,10 +448,16 @@ func mkLocalProxy(ctx context.Context, opts localProxyOpts) (*alpnproxy.LocalPro
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	protocols := append([]alpncommon.Protocol{alpnProtocol}, opts.protocols...)
+	if alpncommon.HasPingSupport(alpnProtocol) {
+		protocols = append([]alpncommon.Protocol{alpncommon.ProtocolWithPing(alpnProtocol)}, protocols...)
+	}
```

Comment:
Currently, we're letting callers define the protocols list without modifying it or validating it, and I just wanted to keep it the same behavior. In addition, I'm considering the ping protocol an "opt-in" feature, where you should specify its usage. That said, it makes more sense to let callers know about it instead of hiding it inside the `alpnproxy` package.

In future releases, if we decide that this is the default behavior, I agree that it should be transparent for callers. (But in this case, we might end up removing the suffix and making the databases protocols already act as a ping protocol).